pkg/types/query/key_filter.go

```diff
@@ -0,0 +1,164 @@
+package query
+
+import (
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+)
+
+// IndexedSequencesKeyFilter creates a KeyFilter that filters logs for the provided sequence property values at the
+// specified property name. Sequence value filters are 'OR'ed together. A sequence read name is the value that
+// identifies the sequence type. The signature value name is the sequence property to apply the filter to and the
+// sequence values are the individual values to search for in the provided property.
+func IndexedSequencesKeyFilter(
+	readName string,
+	propertyName string,
+	values []string,
+	confidence primitives.ConfidenceLevel,
+) KeyFilter {
+	return KeyFilter{
+		Key: readName,
+		Expressions: []Expression{
+			filtersForValues(propertyName, values),
+			Confidence(confidence),
+		},
+	}
+}
```

Comment:
What do you think about making these methods on `KeyFilter`?
```suggestion
func (f *KeyFilter) IndexedSequences(
	propertyName string,
	values []string,
	confidence primitives.ConfidenceLevel,
) {
	f.Expressions = []Expression{
		filtersForValues(propertyName, values),
		Confidence(confidence),
	}
}
```
This leads to shorter func names names and fewer arguments (and keeps the package namespace cleaner)
```go
filter := IndexedSequencesKeyFilter(name, property, values, confidence)
```
```go
filter := KeyFilter{ReadName: name}
filter.IndexedSequences(property, values, confidence)
```
Perhaps the method name could be refined further :shrug:

Reply:
These are intended to only be helper functions that should produce the same queries as the ones in log poller. I would expect the usage as such:

```
contract := types.BoundContract{Address: "0x", Name: "contract"}
seq, err := relay.QueryKey(ctx, contract, IndexedSequencesKeyFilter(readName, property, values, confidence), limiter, retVal)
```

The method name is long, but the original was something like `SelectIndexedLogs` so we keep with the same pattern for the sake of familiarity.

Reply:
I don't think that context changes anything except perhaps to prefer the fluent version.

Reply:
I think I agree with @jmank88 on this, including KeyFilter in every function name is a little confusing so making it a method seems like it might be slightly preferable. But I don't have a strong preference so fine with it either way.